### DISCUSSION TITLE
Size of ground truth images in the multicoil brain test set

### DISCUSSION THREAD

**Discussion by mersad95zd** [facebookresearch/fastMRI#274] [General]
Hi,

Thanks for maintaining this forum. The ground-truth data for the multicoil brain test set is not to be released, but is it possible that you release the ground-truth image size for each test sample?

I'm asking because to save memory during training, one (if not resorting to another option) has to crop input samples. One way is to crop according to the ground-truth image size. In this case, we need ground-truth image sizes for the test set because if we don't crop at inference, a performance drop occurs (due to resolution mismatch).

An alternative would be cropping to the maximum image size of the test set. So if you can't release all image sizes, releasing the maximum size of the test set would also be appreciated. Thanks in advance!

**Comment by mmuckley** [comment]
Hello @mersad95zd, this information is already in the dataset in the HDF5 `attrs` field and is returned to you as `batch.crop_size`. You can see its usage here in the standard `LightningModule`: https://github.com/facebookresearch/fastMRI/blob/ce9a8ef7bd3850eb2688b0cd9530bd5fb91524c3/fastmri/pl_modules/varnet_module.py#L131-L137

This shows how it is extracted from the file:
https://github.com/facebookresearch/fastMRI/blob/ce9a8ef7bd3850eb2688b0cd9530bd5fb91524c3/fastmri/data/transforms.py#L353

There is some extra handling because some FLAIR203 images from the dataset had incorrect cropping information in their headers.

**Reply by mersad95zd** [reply]
Thanks a lot @mmuckley for your prompt response! I totally missed that, especially these lines:

https://github.com/facebookresearch/fastMRI/blob/ce9a8ef7bd3850eb2688b0cd9530bd5fb91524c3/fastmri/data/mri_data.py#L352-L356

**Reply by mmuckley** [reply]
Ah yes - I might have retrieved from there instead. Sorry, it's been awhile since I wrote this code :).

**Comment by tasik34** [comment]
Unfortunately, we can't release specific ground-truth image sizes check correct images dimensions on the [website](https://realiqtestonline.com/). However, the maximum image size for the test set is 320x320. Hope this helps with your memory-saving approach!

**Reply by mmuckley** [reply]
This is not correct - the image size data is available in the metadata, as outlined in my earlier answer.

**Comment by mersad95zd** [comment]
Thanks for the follow-up! My issue was addressed by @mmuckley 's comment `2 years` ago! I'll close it now (should've done so back then).
